### PR TITLE
feature: add image inspect api and cli command

### PR DESCRIPTION
**Pull request by 0x04C2**
Signed-off-by: Wei Fu <fhfuwei@163.com>

**1.Describe what this PR did**

Add image inspect api and cli command

**2.Does this pull request fix one issue?** 
None

**3.Describe how you did it**

Following the development guide, I add:

server side, 

- add `getImage` handler in `image_bridge.go` and router in `router.go`
- for the swagger api doc, I also add `/images/{name}/json` description in swagger.yml

client side,
- add ImageInspect method for api client

cli,

Before this commit, the `image.go` file is used for `images command`. 

`pouch image`, which is parent command, is used to manage pouch image, something like, `rm`, `inspect`, `prune` and `ls` child commands. Based on this,  `image.go` is for `image` command and `image_list.go` is for `images` command. Just rename the file.

**4.Describe how to verify it**

=> image manage command
```
➜  pouch image
pouch image
Manage Pouch image

Usage:
  pouch image [command]

Available Commands:
  inspect     Display detailed information on one image

Flags:
  -h, --help   help for image

Global Flags:
  -H, --host string        Specify connecting address of Pouch CLI (default "unix:///var/run/pouchd.sock")
      --tlscacert string   Specify CA file of TLS
      --tlscert string     Specify cert file of TLS
      --tlskey string      Specify key file of TLS
      --tlsverify          Use TLS and verify remote

Use "pouch image [command] --help" for more information about a command.
```

=> pouch image inspect -h
```
➜  pouch image inspect -h
Return detailed information on Pouch image

Usage:
  pouch image inspect image [flags]

Examples:
$ pouch image inspect docker.io/library/busybox
{
  "CreatedAt": "2017-12-21 04:30:57",
  "Digest": "sha256:bbc3a03235220b170ba48a157dd097dd1379299370e1ed99ce976df0355d24f0",
  "ID": "bbc3a0323522",
  "Name": "docker.io/library/busybox:latest",
  "Size": 720019,
  "Tag": "latest"
}

Flags:
  -h, --help   help for inspect

Global Flags:
  -H, --host string        Specify connecting address of Pouch CLI (default "unix:///var/run/pouchd.sock")
      --tlscacert string   Specify CA file of TLS
      --tlscert string     Specify cert file of TLS
      --tlskey string      Specify key file of TLS
      --tlsverify          Use TLS and verify remote
```

=> inspect real one
```
➜ pouch image inspect docker.io/library/busybox
{
  "CreatedAt": "2017-12-21 04:30:57",
  "Digest": "sha256:bbc3a03235220b170ba48a157dd097dd1379299370e1ed99ce976df0355d24f0",
  "ID": "bbc3a0323522",
  "Name": "docker.io/library/busybox:latest",
  "Size": 720019,
  "Tag": "latest"
}

➜  pouch image inspect docker.io/library/busybox:12
Error: failed to inspect image: {"message":"image: docker.io/library/busybox:12: not found"}

➜  echo $?
1
```

**5.Special notes for reviews**

Test case will be added by the following PR.

### CODE REVIEW

**Comment by 0x04C2** [comment]
ping @allencloud

File: apis/server/image_bridge.go [hunk 1]
```diff
@@ -53,6 +53,17 @@ func (s *Server) listImages(ctx context.Context, resp http.ResponseWriter, req *
 	return json.NewEncoder(resp).Encode(imageList)
 }
 
+func (s *Server) getImage(ctx context.Context, resp http.ResponseWriter, req *http.Request) error {
+	idOrRef := mux.Vars(req)["name"]
+
+	image, err := s.ImageMgr.GetImage(ctx, idOrRef)
+	if err != nil {
+		logrus.Errorf("failed to get image in containerd: %v", err)
```
Comment (allencloud) [inline apis/server/image_bridge.go]: I do not think we should tell users details about `containerd` in pouch daemon's log. And there are others cases that errors are not from containerd.

File: apis/server/image_bridge.go [hunk 2]
```diff
@@ -53,6 +53,17 @@ func (s *Server) listImages(ctx context.Context, resp http.ResponseWriter, req *
...
+		return err
+	}
```
Comment (allencloud) [inline apis/server/image_bridge.go]: Could you add status code 200 here explicitly?
Although by default it is 200.

File: apis/server/image_bridge.go (hunk 1 above)
Comment (0x04C2) [reply apis/server/image_bridge.go]: Updated. I also remove the `containerd` in listImages.

File: apis/server/image_bridge.go (hunk 2 above)
Comment (0x04C2) [reply apis/server/image_bridge.go]: Updated

**Comment by allencloud** [comment]
> image.go is for image command and image_list.go is for images command. Just rename the file.

Reasonable.

**Pull request merged**